t/api/04-jobs.t

```diff
@@ -1222,6 +1225,39 @@ subtest 'show job modules execution time' => sub {
     is($testresults[0]->{execution_time},     '', 'the old format json file does not include execution_time');
 };
 
+subtest 'marking job as done' => sub {
+    # mark running job as done
+    $schema->resultset('Jobs')->find(99961)->update(
+        {
+            state  => RUNNING,
+            result => NONE,
+            reason => undef
+        });
+    $t->post_ok('/api/v1/jobs/99961/set_done?result=incomplete&reason=test')->status_is(200);
+    $t->get_ok('/api/v1/jobs/99961')->status_is(200);
+    my $json = $t->tx->res->json;
+    my $ok   = is($json->{job}->{result}, INCOMPLETE, 'result set');
+    $ok = is($json->{job}->{reason}, 'test', 'reason set') && $ok;
+    $ok = is($json->{job}->{state},  DONE,   'state set')  && $ok;
+    diag explain $json unless $ok;
+
+    # mark job which is already done as done, overriding existing reason
```

Comment:
I prefer to use this comment within the definition of a subtest, e.g.

```perl
subtest 'mark job which is already done as done, overriding existing reason' => sub {
…
}
```

Reply:
done